### PR TITLE
test(playwright): complete page type tests

### PR DESCRIPTION
**Pull request by angeloashmore**
## Context

Completes DT-1835

## The Solution

- Test deleting a page type.
- Test for the blank slate message when the project has no page types.

## Impact / Dependencies

N/A

## Checklist before requesting a review

- [x] I hereby declare my code ready for review.
- [x] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: playwright/tests/pageTypes/pageTypesTable.spec.ts
Comment (xrutayisire) [inline playwright/tests/pageTypes/pageTypesTable.spec.ts]: Test don't pass 👀.

But I was thinking to test closer to the user without testId, maybe we should more verify the blank slate description and button? With that no need to a testId for the blankSlate, WDYT?

We could also check that clicking on the button open the create modal so we ensure user is not blocked on the blank slate forever. WDYT?

**Review by xrutayisire — CHANGES_REQUESTED** [review]
Super cool to continue tech debt roadmap and coverage here!
I added two propositions of changes, let me know.

Note: Baptiste merged a big PR that is super cool removing the need of test.run, so you have conflicts and need to update your new tests. It's really better and exactly how we are supposed to do in Playwright. Baptiste updated the README so you should quickly get it. If not, don't hesitate to update the README also 🙂

**Comment by bapmrl** [comment]
> Note: Baptiste merged a big PR that is super cool removing the need of test.run, so you have conflicts and need to update your new tests.

You just have to replace `test.run()` with `test` @angeloashmore 🙂

File: playwright/tests/pageTypes/pageTypesTable.spec.ts
Comment (angeloashmore) [reply playwright/tests/pageTypes/pageTypesTable.spec.ts]: Oops, it passed locally because I manually deleted the existing custom type before I started. 👀

Testing the message and button is a good idea. I think we still need a test ID to locate the blank slate message and button, however, unless you have a different idea of how to target them.

Here is the updated test: https://github.com/prismicio/slice-machine/blob/cdc4166cb32958340a8a7afedb391741cc1cb6f6/playwright/tests/pageTypes/pageTypesTable.spec.ts#L142-L156

And the new test verifying the blank slate's action button: https://github.com/prismicio/slice-machine/blob/cdc4166cb32958340a8a7afedb391741cc1cb6f6/playwright/tests/pageTypes/pageTypesTable.spec.ts#L158-L178

Comment (xrutayisire) [reply playwright/tests/pageTypes/pageTypesTable.spec.ts]: I'm never sure if we really need to locate the item first. It's more if the text can be at two different places why not. But here for example it seems the description and create button can only be there within the blank slate. It's even more powerful with the id and I don't have a strong opinion so LGTM 🙂

Comment (xrutayisire) [reply playwright/tests/pageTypes/pageTypesTable.spec.ts]: Btw relative to the todo you wrote. I think I may have fixed the problem so you may not need that overhead and the redirection should be perfect! Thanks to the removal of the saga, I forgot about these todo! 👀 Could you check if there are required and remove them if not?

Comment (angeloashmore) [reply playwright/tests/pageTypes/pageTypesTable.spec.ts]: The description is probably fine to check if it is present anywhere on the page, but the "Create" button could be confused with the non-blank-slate "Create" button. We don't currently show two Create buttons, but I'm thinking we should be explicit about which button we are checking for in case we have a regression somewhere.

Regarding the TODO: You are totally right! I tested it without the extra navigation before, and it failed. But now it works without it. There must have been something else wrong with the test back then.

I'll remove the other uses of `goto()`.

Comment (xrutayisire) [reply playwright/tests/pageTypes/pageTypesTable.spec.ts]: In this case I guess it's nice to be specific yes as we know we have another button with the same label, so all good with the testid 👍

Super cool! Something less to take care haha

**Review by xrutayisire — APPROVED** [review]
LGTM after the changes we discussed about 🙂

**Pull request merged**
